src/Orleans/Serialization/SerializationTestEnvironment.cs

```diff
@@ -0,0 +1,34 @@
+namespace Orleans.Serialization
+{
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    using Orleans.Runtime;
+
+    public class SerializationTestEnvironment
+    {
+        public SerializationTestEnvironment()
+        {
+            this.TypeMetadataCache = new TypeMetadataCache();
+            this.AssemblyProcessor = new AssemblyProcessor(this.TypeMetadataCache);
+            this.GrainFactory = new GrainFactory(null, this.TypeMetadataCache);
+        }
+
+        public void InitializeForTesting(List<TypeInfo> serializationProviders = null, TypeInfo fallbackType = null)
+        {
+            SerializationManager.InitializeForTesting(serializationProviders, fallbackType);
+            this.AssemblyProcessor.Initialize();
+        }
+
+        public static SerializationTestEnvironment Initialize(List<TypeInfo> serializationProviders = null, TypeInfo fallbackType = null)
```

Comment:
>List<TypeInfo> serializationProviders = null, TypeInfo fallbackType = null [](start = 62, length = 74)

I see this being called with and explicit Initialize(null, null) most of the times. Shouldn't they use the default optional parameters implicitly instead when tests do not really care about these values at all?